tools/monitor/backend.py

```diff
@@ -0,0 +1,193 @@
+from telecommand.fs import DownloadFile 
+
+from gui import MonitorUI
+import locale
+
+
+class MonitorBackend:
+    def __init__(self):
+        self.Decoder = response_frames.FrameDecoder(response_frames.frame_factories)
+        self.abort_push = zmq.Context.instance().socket(zmq.PAIR)
+        self.abort_push.bind('inproc://download_monitor/abort')
+        self.abort_pull = zmq.Context.instance().socket(zmq.PAIR)
+        self.abort_pull.connect('inproc://download_monitor/abort')
+
+    def parse_args(self):
+        parser = argparse.ArgumentParser()
+        parser.add_argument('session', nargs=1, help='Session number')
+        parser.add_argument('address', nargs='+', help='Address to connect for frames')
+        parser.add_argument('--port', '-p', help='Command port', type=int, default=7007)
+        parser.add_argument('--mission', '-m', help='Mission repo', default=os.path.join(os.path.dirname(__file__), '..', '..', '..', 'mission'))
+
+        return parser.parse_args()
+
+    def connect_sockets(self, addresses):
+        ctx = zmq.Context.instance()
+        result = []
+
+        for a in addresses:
+            socket = ctx.socket(zmq.SUB)
+            socket.setsockopt(zmq.SUBSCRIBE, '')
+            socket.connect(a)
+            result.append(socket)
+
+        return result
+
+    def parse_frame(self, raw_frame):
+        return self.Decoder.decode(ensure_byte_list(raw_frame[16:-2]))
+
+    def load_tasklist(self, mission_path, session):
+        tasks_file_path = os.path.join(mission_path, 'sessions', str(session), 'tasklist.py')  
+    
+        import telecommand as tc
+        import datetime
+        from radio.task_actions import Send, SendLoop, Sleep, Print, SendReceive, WaitMode
+        from devices import camera
+        from devices.adcs import AdcsMode
+        from devices.camera import CameraLocation, PhotoResolution
+        from devices.comm import BaudRate
+        
+        if not os.path.exists(tasks_file_path):
+            print("Could not find file {}".format(tasks_file_path))
+            return []
+
+        with open(tasks_file_path) as tasks_file:
+            exec(tasks_file)
+            tasks_file.close()
+
+        return tasks
+
+    def create_dictionary(self, tasklist):
+            commandsDict = OrderedDict()
+
+            for item in tasklist:
+                command = item[0]
+                if not isinstance(command, DownloadFile):
+                    continue
+                commandsDict[command._correlation_id] = command
+
+            return commandsDict
+
+    def generate_new_tasklist(self, commandsDict):
+        newTasks = []
+        for _, command in commandsDict.items():
+            newTasks.append(command)
+
+        return newTasks
+
+    def process_frame(self, frame, commandsDict, ui):
+        if not isinstance(frame, response_frames.common.FileSendSuccessFrame) and not isinstance(frame, response_frames.common.FileSendErrorFrame):
```

Comment:
`if not type(frame) in [A,B]:` trochę zwięźlej :)

Reply:
Zmienione.